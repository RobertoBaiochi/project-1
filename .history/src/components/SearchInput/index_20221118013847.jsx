import React from 'react'

const SearchInput = ({} ) => {
  return (
    <input
          value={searchValue}
          onChange={this.handleChange} 
          type="search" 
        />
  )
}

export default SearchInput