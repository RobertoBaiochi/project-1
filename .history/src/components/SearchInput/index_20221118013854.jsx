import React from 'react'

const SearchInput = ( {searchValue, han} ) => {
  return (
    <input
          value={searchValue}
          onChange={this.handleChange} 
          type="search" 
        />
  )
}

export default SearchInput