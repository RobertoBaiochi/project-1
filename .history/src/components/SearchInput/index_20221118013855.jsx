import React from 'react'

const SearchInput = ( {searchValue, handleChange} ) => {
  return (
    <input
          value={searchValue}
          onChange={this.handleChange} 
          type="search" 
        />
  )
}

export default SearchInput