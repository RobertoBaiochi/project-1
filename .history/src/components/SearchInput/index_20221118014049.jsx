import './styles.css'

const SearchInput = ( {searchValue, handleChange} ) => {
  return (
    <input
          value={searchValue}
          onChange={this.handleChange} 
          type="search" 
    />
  )
}

export default SearchInput