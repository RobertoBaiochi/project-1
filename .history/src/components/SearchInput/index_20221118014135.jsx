import './styles.css'

const SearchInput = ( {searchValue, handleChange} ) => {
  return (
    <input
      value={searchValue}
      onChange={handleChange} 
      type="search" 
    />
  )
}

export default SearchInput