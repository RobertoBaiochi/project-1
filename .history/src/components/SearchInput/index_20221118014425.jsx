import './styles.css'

const SearchInput = ( {searchValue, handleChange} ) => {
  return (
    <input
      className=''
      value={searchValue}
      onChange={handleChange} 
      type="search" 
    />
  )
}

export default SearchInput