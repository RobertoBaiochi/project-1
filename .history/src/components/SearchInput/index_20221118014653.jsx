import './styles.css'

const SearchInput = ( {searchValue, handleChange} ) => {
  return (
    <input
      className='search-input'
      value={searchValue}
      onChange={handleChange} 
      type="search" 
      
    />
  )
}

export default SearchInput