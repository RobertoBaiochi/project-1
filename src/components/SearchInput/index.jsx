import P from 'prop-types';

import './styles.css';

const SearchInput = ({ searchValue, handleChange }) => {
  return (
    <input
      className="search-input"
      value={searchValue}
      onChange={handleChange}
      type="search"
      placeholder="Type your search..."
    />
  );
};

export default SearchInput;

SearchInput.propTypes = {
  searchValue: P.string.isRequired,
  handleChange: P.func.isRequired,
};
