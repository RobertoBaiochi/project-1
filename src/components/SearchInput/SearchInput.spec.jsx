import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SearchInput from '.';
import TestRenderer from 'react-test-renderer';

describe('<SearchInput />', () => {
  it('should have a value on searchValue', () => {
    const fn = jest.fn();
    render(<SearchInput handleChange={fn} searchValue={'testing input'} />);

    const input = screen.getByPlaceholderText(/type your search/i);

    expect(input.value).toBe('testing input');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<SearchInput handleChange={fn} searchValue={'um valor qualquer'} />);

    const input = screen.getByPlaceholderText(/type your search/i);

    const value = 'o valor';

    userEvent.type(input, value);

    expect(input.value).toBe('um valor qualquer');
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();

    const testRenderer = TestRenderer.create(<SearchInput handleChange={fn} searchValue={''} />).toJSON();

    expect(testRenderer).toMatchSnapshot();
  });
});
