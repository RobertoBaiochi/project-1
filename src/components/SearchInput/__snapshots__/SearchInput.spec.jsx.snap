// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should match snapshot 1`] = `
<input
  className="search-input"
  onChange={[MockFunction]}
  placeholder="Type your search..."
  type="search"
  value=""
/>
`;
